file.c

```diff
@@ -276,12 +349,104 @@ int kibosh_release(const char *path UNUSED, struct fuse_file_info *info)
     return AS_FUSE_ERR(ret);
 }
 
+_Pragma("GCC diagnostic push")
+_Pragma("GCC diagnostic ignored \"-Wdiscarded-qualifiers\"")
```

Comment:
Is this needed?

Reply:
When corrupting write calls, the write buf needed to be changed in order to write corrupted data to disk. However, the buf pointer is a const, so changing it will raise a discarded-qualifier warning from the compiler. The pragma is only to suppress the warning during compilation.

Reply:
I'd really rather not modify a const buffer.  It's technically undefined behavior.  Let's just copy the buffer in the case where we need to do this.  You should be able to copy it on the stack since it's a few KB at most